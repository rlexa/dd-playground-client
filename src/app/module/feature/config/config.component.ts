import {CommonModule} from '@angular/common';
import {ChangeDetectionStrategy, Component} from '@angular/core';
import {MatCardModule} from '@angular/material/card';
import {MatListModule} from '@angular/material/list';

@Component({
  selector: 'app-config',
  template: `<mat-card appearance="outlined">
    <mat-card-title>Configuration</mat-card-title>
    <mat-card-content>
      <mat-list class="mat-list-auto-height">
        <h3 mat-subheader>Nothing to configure yet.</h3>
      </mat-list>
    </mat-card-content>
  </mat-card>`,
  changeDetection: ChangeDetectionStrategy.OnPush,
  standalone: true,
  imports: [CommonModule, MatCardModule, MatListModule],
})
export class ConfigComponent {}
