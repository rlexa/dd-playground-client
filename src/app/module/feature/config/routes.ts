import {Route} from '@angular/router';
import {RouteRoot, RouteWild} from 'src/app/routing';

export default [
  {path: RouteRoot, loadComponent: () => import('./config.component').then((ii) => ii.ConfigComponent)},
  {path: RouteWild, redirectTo: RouteRoot},
] as Route[];
