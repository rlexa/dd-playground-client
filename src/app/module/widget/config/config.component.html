<mat-card appearance="outlined">
  <mat-card-title>Configuration</mat-card-title>
  <mat-card-content>
    <mat-list class="mat-list-auto-height">
      <h3 mat-subheader>Nothing to configure yet.</h3>
    </mat-list>
  </mat-card-content>
</mat-card>
