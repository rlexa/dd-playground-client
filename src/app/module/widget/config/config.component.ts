import { ChangeDetectionStrategy, Component } from '@angular/core';

@Component({
  selector: 'app-config',
  templateUrl: './config.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ConfigComponent { }
