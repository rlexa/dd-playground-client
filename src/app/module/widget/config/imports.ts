import { CommonModule } from '@angular/common';
import { MatCardModule, MatListModule } from '@angular/material';
import { RouterModule } from '@angular/router';

export const imports = [CommonModule, RouterModule, MatCardModule, MatListModule];
