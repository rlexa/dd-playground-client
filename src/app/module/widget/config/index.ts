export * from './config.component';
export * from './config.module';
