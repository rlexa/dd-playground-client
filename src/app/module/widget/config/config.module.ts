import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { ROUTE_ROOT, ROUTE_WILDCARD } from 'app/routing';
import { ConfigComponent } from './config.component';
import { imports } from './imports';

const ROUTING = [
  { path: ROUTE_ROOT, component: ConfigComponent },
  { path: ROUTE_WILDCARD, redirectTo: ROUTE_ROOT, pathMatch: 'full' },
];

@NgModule({
  imports: [...imports, RouterModule.forChild(ROUTING)], exports: [ConfigComponent], declarations: [ConfigComponent]
})
export class ConfigModule { }
