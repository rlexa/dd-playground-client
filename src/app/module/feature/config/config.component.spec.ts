import {ComponentFixture} from '@angular/core/testing';
import {MockBuilder, MockRender} from 'ng-mocks';
import {ConfigComponent} from './config.component';

describe('ConfigComponent', () => {
  let fixture: ComponentFixture<ConfigComponent>;

  beforeEach(() => MockBuilder(ConfigComponent));

  beforeEach(() => {
    fixture = MockRender(ConfigComponent);
    fixture.detectChanges();
  });

  test('renders', () => expect(fixture).toMatchSnapshot());
});
