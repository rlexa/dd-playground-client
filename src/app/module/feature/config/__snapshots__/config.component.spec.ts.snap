// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigComponent renders 1`] = `
<mock-render>
  <app-config>
    <mat-card
      appearance="outlined"
      ng-reflect-appearance="outlined"
    >
      <mat-card-title>
        Configuration
      </mat-card-title>
      <mat-card-content>
        <mat-list
          class="mat-list-auto-height"
        >
          <h3
            mat-subheader=""
          >
            Nothing to configure yet.
          </h3>
        </mat-list>
      </mat-card-content>
    </mat-card>
  </app-config>
</mock-render>
`;
